src/app/mytools/my-tool/my-tool.component.ts

```diff
@@ -57,95 +60,192 @@ export class MyToolComponent implements OnInit {
+      if (tools) {
+        this.tools = tools;
         const sortedContainers = this.mytoolsService.sortNSContainers(tools, this.user.username);
-        this.containerService.setNsContainers(sortedContainers);
-      }
-    });
-    this.containerService.nsContainers.subscribe(containers => {
-      this.nsContainers = containers;
-      /* For the first initial time, set the first tool to be the selected one */
-      if (this.nsContainers && this.nsContainers.length > 0) {
-        const theFirstTool = this.nsContainers[0].containers[0];
-        const foundWorkflow = this.findToolFromPath(this.urlResolverService.getEntryPathFromUrl(), this.nsContainers);
-        if (foundWorkflow) {
-          this.selectContainer(foundWorkflow);
+        /* For the first initial time, set the first tool to be the selected one */
+        if (sortedContainers && sortedContainers.length > 0) {
+          this.orgToolsObject = this.convertNamespaceWorkflowsToOrgWorkflowsObject(sortedContainers);
+          const foundWorkflow = this.findToolFromPath(this.urlResolverService.getEntryPathFromUrl(), this.orgToolsObject);
+          if (foundWorkflow) {
+            this.selectContainer(foundWorkflow);
+          } else {
+            const publishedWorkflow = this.getFirstPublishedEntry(sortedContainers);
+            if (publishedWorkflow) {
+              this.selectContainer(publishedWorkflow);
+            } else {
+              const theFirstWorkflow = sortedContainers[0].containers[0];
+              this.selectContainer(theFirstWorkflow);
+            }
+          }
         } else {
-          this.selectContainer(theFirstTool);
+          this.selectContainer(null);
         }
-      } else {
-        this.selectContainer(null);
       }
     });
     this.stateService.refreshMessage$.subscribe(refreshMessage => this.refreshMessage = refreshMessage);
     this.registerToolService.tool.subscribe(tool => this.registerTool = tool);
   }
 
-  link() {
-      this.accountsService.link(TokenSource.GITHUB);
+  ngOnDestroy() {
+    this.ngUnsubscribe.next();
+    this.ngUnsubscribe.complete();
   }
 
-  private findToolFromPath(path: string, nsContainers: any[]): ExtendedDockstoreTool {
-    let matchingWorkflow: ExtendedDockstoreTool;
-    nsContainers.forEach((nsContainer)  => {
-      nsContainer.containers.forEach((container: ExtendedDockstoreTool) => {
-        if (container.tool_path === path) {
-          matchingWorkflow = container;
+/**
+   * This figures out which tab (Published/Unpublished) is active
+   * In order of priority:
+   * 1. If the selected entry is published/unpublished, the tab selected will published/unpublished to reflect it
+   * 2. If there are published entries, the published tab will be selected
+   * 3. Unpublished otherwise
+   * @private
+   * @memberof MyToolComponent
+   */
+  private updateActiveTab() {
+    if (this.orgToolsObject) {
+      for (let i = 0; i < this.orgToolsObject.length; i++) {
```

Comment:
Is it possible to clean up the array indices `[i]` with an iterator or similar ?
https://www.typescriptlang.org/docs/handbook/iterators-and-generators.html

Reply:
Ran into problems trying that, went with this guy's opinion instead https://stackoverflow.com/a/3010848/4484759

Reply:
Gross. Boo@typescript